geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -162,7 +171,10 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val cfg = new IteratorSetting(iteratorPriority_SpatioTemporalIterator,
       "within-" + randomPrintableString(5),classOf[IndexIterator])
 
-    configureStFilter(cfg, filter)
+    // test if the filter is for the whole world
+    val isWholeWorld = filter.exists(isFilterWholeWorld)
+    if (!isWholeWorld) configureStFilter(cfg, filter)
```

Comment:
@anthonyccri @AndrewAnnex This is a great question.  

tl;dr: I think the check is unnecessary and should be removed.

Check out this: https://github.com/jnh5y/geomesa/commit/9ca8d2657b0d10a534cb19ebefe017a3db29ca1c

If there is a whole world clause, it has been removed by this point.